Cannot find module 'pretty-format'

- `@testing-library/react` version: 10.2.0
- Testing Framework and version:
  * `@testing-library/dom` version 7.9.0
  * `jest` version 25.5.4
- DOM Environment:
  * `jsdom` version 15.2.1
  * `react` version 16.13.1
  * `node` version 13.6.0
  * `typescript` version 3.9.3

### What you did:

Dependabot upgrade of `@testing-library/react` from 10.0.4 to 10.2.0.

### What happened:

In CI, our npm script that invokes `tsc --noEmit` fails with the following:

```
➜  assets git:(dependabot/npm_and_yarn/assets/testing-library/react-10.2.0) npx tsc --noEmit
node_modules/@testing-library/react/types/index.d.ts:3:54 - error TS2307: Cannot find module 'pretty-format' or its corresponding type declarations.

3 import {OptionsReceived as PrettyFormatOptions} from 'pretty-format'
                                                       ~~~~~~~~~~~~~~~
```

### Problem description:

It can't find `pretty-format`. I do see `pretty-format` in my `package-lock.json`, version 25.5.0. It might be the "corresponding type declarations" that the error is flagging? I notice this line was added in #690, which finagles the type definitions a bit, though I don't totally understand the PR.

It could be that I'm supposed to update my package-json with a new dependency of some sort?

I'm getting the same error:

```
node_modules/@testing-library/react/types/index.d.ts:3:54 - error TS2307: Cannot find module 'pretty-format' or its corresponding type declarations.

3 import {OptionsReceived as PrettyFormatOptions} from 'pretty-format'
                                                       ~~~~~~~~~~~~~~~
```

I believe it's because it's listed as a dev dependency, which means that when I run `npm install` from _my_ root, it won't install it. When I run `npm install` it will install `dependencies` and `devDependencies` for me, but not `devDependencies` **of** my dependencies.

I note that in `dom-testing-library`, on which your PR was based, includes it [as a dependency](https://github.com/testing-library/dom-testing-library/blob/281eb8b62700dec9a79ad315b6055a61f61f3c9f/package.json). Perhaps you need to do the same here?

> I'm getting the same error:
> 
> ```
> node_modules/@testing-library/react/types/index.d.ts:3:54 - error TS2307: Cannot find module 'pretty-format' or its corresponding type declarations.
> 
> 3 import {OptionsReceived as PrettyFormatOptions} from 'pretty-format'
>                                                        ~~~~~~~~~~~~~~~
> ```
> 
> I believe it's because it's listed as a dev dependency, which means that when I run `npm install` from _my_ root, it won't install it. When I run `npm install` it will install `dependencies` and `devDependencies` for me, but not `devDependencies` **of** my dependencies.
> 
> I note that in `dom-testing-library`, on which your PR was based, includes it [as a dependency](https://github.com/testing-library/dom-testing-library/blob/281eb8b62700dec9a79ad315b6055a61f61f3c9f/package.json). Perhaps you need to do the same here?

I've updated the branch to include it as a Dep vs devDep but we're discussing whether or not it's really necessary. We would think `pretty-format` should be installed from `@testing-library/dom` since it already lists it as a dependency.

Getting similar error with yarn2. When we run `tsc`. It fails with 
``` error TS2307: Cannot find module 'pretty-format' or its corresponding type declarations.

3 import {OptionsReceived as PrettyFormatOptions} from 'pretty-format'```

--- a/package.json
+++ b/package.json
@@ -44,7 +44,7 @@
   "license": "MIT",
   "dependencies": {
     "@babel/runtime": "^7.12.1",
-    "@testing-library/dom": "^7.26.4"
+    "@testing-library/dom": "^7.26.6"
   },
   "devDependencies": {
     "@testing-library/jest-dom": "^5.11.5",

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,7 +1,11 @@
 // TypeScript Version: 3.8
 
-import {OptionsReceived as PrettyFormatOptions} from 'pretty-format'
-import {queries, Queries, BoundFunction} from '@testing-library/dom'
+import {
+  queries,
+  Queries,
+  BoundFunction,
+  prettyFormat,
+} from '@testing-library/dom'
 import {act as reactAct} from 'react-dom/test-utils'
 
 export * from '@testing-library/dom'
@@ -15,7 +19,7 @@ export type RenderResult<Q extends Queries = typeof queries> = {
       | DocumentFragment
       | Array<HTMLElement | DocumentFragment>,
     maxLength?: number,
-    options?: PrettyFormatOptions,
+    options?: prettyFormat.OptionsReceived,
   ) => void
   rerender: (ui: React.ReactElement) => void
   unmount: () => boolean

